### PR TITLE
Feat: global zoom for all charts

### LINKED ISSUE #409: chartings - zoom
## Detailed Description
Investigate if it's possible to add an option to zoom in on the national, gsp, and pv site forecast charts. 

## Context
This is a feature we have on the `OCF Dashboard` thanks to the plotly library and it's  possible to click on a chart and you are able to zoom in and get a bit more granularity on the chart. 

## Possible Implementation

- [ ] Investigate Rechart’s functionality: [recharts library](https://recharts.org/en-US/examples/HighlightAndZoomLineChart) 
- [ ] Investigate a different package
- [ ] Talk to users

### PR DESCRIPTION
**Pull request by rachel-labri-tipton**
# Pull Request

## Description

Current code for chart with zoom function that has access to all charts. 

Fixes #409 

## How Has This Been Tested?

Ran the code locally. 

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by rachel-labri-tipton** [comment]
hey @braddf, it looks like the zoom has deployed ok. I clicked around and everything seems to be working on the deployed branch.  would it be possible to have a look at it and let me know if you're good if it get's merged to `dev`?

It seems the quartz-frontend has failed to deploy, but will this impact merging these changes to `dev`?

File: apps/nowcasting-app/components/charts/remix-line.tsx [hunk 1]
```diff
@@ -197,20 +213,68 @@ const RemixLine: React.FC<RemixLineProps> = ({
...
+      setFilteredPreppedData(dataInAreaRange);
+    }
+  }, [globalZoomArea, globalIsZooming, preppedData, zoomEnabled]);
+
   return (
     <div style={{ position: "relative", width: "100%", height: "100%" }}>
+      {zoomEnabled && globalIsZoomed && (
+        <div
+          className={
+            deltaView ? `absolute top-5 right-16 mr-3 z-10` : `absolute top-5 right-4 z-10`
```
Comment (braddf) [inline apps/nowcasting-app/components/charts/remix-line.tsx:251]: If you could separate out the shared classes from this ternary that would be great, putting them at the beginning and then using a template string for the conditional classes, keeps things neat and DRY

File: apps/nowcasting-app/components/charts/remix-line.tsx [hunk 2]
```diff
@@ -197,20 +213,68 @@ const RemixLine: React.FC<RemixLineProps> = ({
...
-          data={preppedData}
+          data={zoomEnabled && globalIsZoomed ? filteredPreppedData : preppedData}
           margin={{
             top: 20,
             right: 16,
             bottom: 20,
             left: 16
           }}
           onClick={(e?: { activeLabel?: string }) => {
+            console.log("click onclick clicked", e?.activeLabel, timeOfInterest);
```
Comment (braddf) [inline apps/nowcasting-app/components/charts/remix-line.tsx:277]: Console

File: apps/nowcasting-app/components/charts/remix-line.tsx [hunk 3]
```diff
@@ -219,6 +283,44 @@ const RemixLine: React.FC<RemixLineProps> = ({
                 : setTimeOfInterest(e.activeLabel);
             }
           }}
+          onMouseDown={(e?: { activeLabel?: string }) => {
+            console.log("click onmousedown clicked");
```
Comment (braddf) [inline apps/nowcasting-app/components/charts/remix-line.tsx:287]: Console

File: apps/nowcasting-app/components/icons/icons.tsx [hunk 4]
```diff
@@ -51,6 +51,22 @@ export const CloseButtonIcon: React.FC<IconProps> = ({ className }) => (
   </svg>
 );
 
+export const CloseButtonIconForZoom: React.FC<IconProps> = ({ className }) => (
```
Comment (braddf) [inline apps/nowcasting-app/components/icons/icons.tsx:54]: Don't think we're using this X icon now?

**Review by braddf — CHANGES_REQUESTED** [review]

**Comment by rachel-labri-tipton** [comment]
hey @braddf, thanks for the feedback. I made the requested changes.

**Review by braddf — APPROVED** [review]
LGTM!

**Pull request merged**